docs/protocol/specifications/vm1.md

```diff
@@ -75,8 +75,8 @@ Execution of any of the following instructions results in immediate failure:
 * [AMOUNT](#amount)
 * [MINTIME](#mintime)
 * [MAXTIME](#maxtime)
-* [TXREFDATAHASH](#txrefdatahash)
-* [REFDATAHASH](#refdatahash)
+* [TXDATAHASH](#txdatahash)
```

Comment:
>  whether or not this is a hash is an app-level decision

The spec says "Data is a Hash field type". I understand that it could not be a hash, but just some 32-byte string, but then we should probably redefine that field. However, I'd rather not define it as varstring to highlight that any actual data longer than 32 bytes should go somewhere else in the transport layer.